File: Model/Entity/Recipe.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using EFGetStarted.Attributes;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace EFGetStarted.Model.Entity;

public class Recipe : AbstractEntity
{
    [Required]
    [RecipeCodeValidation]
    public string Code { get; set; }
    [Required]
    public string Name { get; set; }
    [Required]
    public string Description { get; set; }
    public List<IngredientGroup> IngredientGroups { get; set; }
    public long PreparationDuration { get; set; }
    [Required]
    public long CookingDuration { get; set; }
    
    [Required]
    public int CreatedByUserId { get; set; }
    
    [ForeignKey("CreatedByUserId")]
    public ApplicationUser CreatedBy { get; set; }
    
    public string? ImgBase64 { get; set; }

    public long TotalDuration => PreparationDuration + CookingDuration;
    
    
}

public class RecipeTypeConfiguration : IEntityTypeConfiguration<Recipe>
{
    public void Configure(EntityTypeBuilder<Recipe> builder)
    {
        builder.HasQueryFilter(e => !e.Deleted);
    }
}

File: src/main/kotlin/com/example/recipecollection/domain/Recipe.kt
package com.example.recipecollection.domain

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

@Entity
@Table(name = "recipes")
class Recipe(
    @field:NotBlank @Column(nullable = false) var code: String,
    @field:NotBlank @Column(nullable = false) var name: String,
    @field:NotBlank @Column(nullable = false, columnDefinition = "TEXT") var description: String,
    @Column(nullable = false) var preparationDuration: Long = 0,
    @field:NotNull @Column(nullable = false) var cookingDuration: Long,
    @field:NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id", nullable = false)
    var createdBy: ApplicationUser,
    @Column(columnDefinition = "TEXT") var imgBase64: String? = null,
) : AbstractEntity() {
    @OneToMany(mappedBy = "recipe", cascade = [CascadeType.ALL], orphanRemoval = true)
    var ingredientGroups: MutableList<IngredientGroup> = mutableListOf()

    val totalDuration: Long
        get() = preparationDuration + cookingDuration
}
